src/lcm/dispatchers.py

```diff
@@ -1,24 +1,34 @@
 import inspect
+from collections.abc import Callable
+from typing import TypeVar
 
 from jax import vmap
 
 from lcm.functools import allow_args, allow_kwargs
 
+F = TypeVar("F", bound=Callable)
 
-def spacemap(func, dense_vars, sparse_vars, *, dense_first):
+
+def spacemap(
+    func: F,
+    dense_vars: list[str],
+    sparse_vars: list[str],
+    *,
+    dense_first: bool,
+) -> F:
     """Apply vmap such that func is evaluated on a space of dense and sparse variables.
 
-    This is achieved by applying a product map for all dense_vars and a vmap for the
-    sparse_vars.
+    This is achieved by applying a product map for all dense variables and a vmap for
+    the sparse variables.
 
     In contrast to vmap, spacemap preserves the function signature and allows the
     function to be called with keyword arguments.
 
     Args:
         func (callable): The function to be dispatched.
-        dense_vars (list): Names of the dense variables, i.e. those that are simply
-            stored as array of possible values in the grid because the possible values
-            of those variable does not depend on the value of other variables.
+        dense_vars (list): Names of the dense variables, i.e. those that are stored
+            as arrays of possible values in the grid, because the possible values
```

Comment:
```suggestion
            as arrays of possible values in the grid.
```
Remove rest of sentence, too. Nobody should look here to understand the concept. Could add link to dev docs or whatever the right place is, though.